addons/mail/static/src/components/message_list/message_list.js

```diff
@@ -224,7 +228,12 @@ class MessageList extends Component {
                 )
             )
             .map(([refId, ref]) => ref)
-            .sort((ref1, ref2) => (ref1.message.id < ref2.message.id ? -1 : 1));
+              .sort((ref1, ref2) => {
+                  if (ref1.message.id < 0 || ref2.message.id < 0) {
+                      return (ref1.message.id < ref2.message.id ? 1 : -1);
+                  }
+                  return (ref1.message.id < ref2.message.id ? -1 : 1);
```

Comment:
probably still wrong? I think it should have a case all positive, a case all negative, and a case one positive and one negative (and its reverse)